Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/openSAML/OpenSAML2ComponentBuilder.java

```diff
@@ -444,29 +342,39 @@ private SubjectConfirmationData createSubjectConfirmationData(X509Certificate ce
      * @param certificate the certificate
      * @param publicKey the public key
      * @return the key info
-     * @throws Exception the exception
+     * @throws SAMLComponentBuilderException the exception
      */
-    public KeyInfo getKeyInfo(X509Certificate certificate, PublicKey publicKey) throws Exception {
-        KeyInfo ki = (KeyInfo) createOpenSAMLObject(KeyInfo.DEFAULT_ELEMENT_NAME);
 
-        BasicX509Credential credential = new BasicX509Credential();
-        credential.setEntityCertificate(certificate);
+    public KeyInfo getKeyInfo(final PublicKey publicKey) throws SAMLComponentBuilderException {
+        final KeyInfoBean keyInfoBean = new KeyInfoBean();
+        KeyInfo keyInfo = null;
+        keyInfoBean.setPublicKey(publicKey);
+        try {
+            keyInfo = SAML1ComponentBuilder.createKeyInfo(keyInfoBean);
+        } catch (SecurityException | WSSecurityException e) {
+            LOG.error(e.getLocalizedMessage(), e);
+            throw new SAMLComponentBuilderException(e.getLocalizedMessage(), e);
+        }
 
-        KeyInfoHelper.addPublicKey(ki, publicKey);
-        // KeyInfoHelper.addCertificate(ki, certificate);
-        return ki;
+        return keyInfo;
     }
 
     /**
      * Gets the public key.
      *
      * @return the public key
-     * @throws Exception the exception
+     * @throws SAMLComponentBuilderException the exception
      */
-    public PublicKey getPublicKey() throws Exception {
-        CertificateManager cm = CertificateManagerImpl.getInstance();
+    public PublicKey getPublicKey() throws SAMLComponentBuilderException {
+        final CertificateManager cm = CertificateManagerImpl.getInstance();
+        X509Certificate certificate = null;
 
-        X509Certificate certificate = cm.getDefaultCertificate();
+        try {
+            certificate = cm.getDefaultCertificate();
+        } catch (final CertificateManagerException e) {
+            LOG.error(e.getLocalizedMessage(), e);
```

Comment:
why log error again when throwing back exception? Caller will be responsible for catching exception and log at their side.